### PR TITLE
Added sorting by strand order for transcript subcomponents.

### LINKED ISSUE #7051: Latest Funcotator DataSources Gencode hg19 has unexpected order for transcript segments
The 1.7 release of the data sources has a lifted over hg19 version of Gencode.

In this version contrary to other releases, the individual elements of each transcript seem to be represented in numerical order, rather than the order in which they appear in the transcript at transcription time.  For `+` strand transcripts, this doesn't matter, but for `-` strand transcripts, the ordering of the exons/CDS regions in the gencode gtf file is reversed to what is expected.  The result is that the coding sequence, protein prediction, and other annotations are incorrect.

One user has already run into this issue: https://gatk.broadinstitute.org/hc/en-us/community/posts/360076207992--Repost-Wrong-annotation-with-Funcotator-1-7

One of two fixes is required:

1. Update the code to always sort the transcript elements by how they appear in the transcribed sequence
2. When generating the Gencode data, always sort the transcript elements by their transcribed order.

We should do both and later roll back the sorting to optimize for speed.

### PR DESCRIPTION
**Pull request by jonn-smith**
- Fixes #7051

### CODE REVIEW

File: src/test/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactoryUnitTest.java [hunk 1]
```diff
@@ -1468,6 +1490,15 @@ public void testCreateFuncotations(final String expectedGeneName,
                                 final ReferenceDataSource referenceDataSource,
                                 final String transcriptFastaFile, final String transcriptGtfFile) {
 
+        //        outList.addAll(
+        //        addReferenceDataToUnitTestData(DataProviderForTroubleVariants.provideTP53AllelesForOrderTestsHg19(),
+        //        FuncotatorReferenceTestUtils.retrieveHg19Chr17Ref(),
+        //        gencodeHg19TP53FeatureReader,
+        //        refDataSourceHg19Ch17,
+        //        FuncotatorTestConstants.TP53_REVERSE_ORDER_GENCODE_TRANSCRIPT_FASTA_FILE,
+        //        FuncotatorTestConstants.TP53_REVERSE_ORDER_GENCODE_TRANSCRIPT_FASTA_FILE ) );
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactoryUnitTest.java:1499]: Why is this commented out?

**Review by droazen — CHANGES_REQUESTED** [review]
@jonn-smith Back to you with a question

File: src/test/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactoryUnitTest.java (hunk 1 above)
Comment (jonn-smith) [reply src/test/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactoryUnitTest.java:1499]: That was some code I forgot to remove.  I had a little trouble determining which fields were being filled in by the helper methods (hence the commented out code at line 195 to remind people making new tests).

I'll remove it.

**Review by droazen — APPROVED** [review]
:+1: Merge when tests pass

**Pull request merged**